src/mdoc/models/device-response.ts

```diff
@@ -137,6 +139,26 @@ export class DeviceResponse extends CborStructure {
 
       await document.issuerSigned.verify({ verificationCallback: onCheck }, ctx)
     }
+
+    if (options.deviceRequest?.docRequests && this.documents) {
+      try {
+        verifyDocRequestsWithIssuerSigned(
+          options.deviceRequest.docRequests,
+          this.documents.map((d) => d.issuerSigned)
+        )
+        onCheck({
+          status: 'PASSED',
+          check: 'Device Response did match the Device Request',
+          category: 'DOCUMENT_FORMAT',
```

Comment:
This is fine for now, but it'd be great if the verify method never throws an error, and instead we will populate a typed context object that will be returned, with e.g. `certificateChainValid`, `signatureValid`, `xxxValid`. So we can also start returning those errors in Credo and Paradym.

Also is it correct that we don't throw after the `catch` at the moment?

Reply:
Ah i see the callback throws if there's an error 👍  Makes sense.